src/session/content/chat_action_bar.rs

```diff
@@ -282,13 +286,218 @@ impl ChatActionBar {
         self.save_message_as_draft();
 
         let imp = self.imp();
+        let mut bindings = imp.bindings.borrow_mut();
+        while let Some(binding) = bindings.pop() {
+            binding.unwatch();
+        }
 
         if let Some(ref chat) = chat {
             self.load_draft_message(chat.draft_message());
 
             imp.chat_action_in_cooldown.set(false);
+
+            let chat_expression = gtk::ConstantExpression::new(chat).upcast();
+            let permissions_expression = gtk::PropertyExpression::new(
+                Chat::static_type(),
+                Some(chat_expression.clone()),
+                "permissions",
+            )
+            .upcast();
+
+            // Handle whether or not message bar should be shown
+            match chat.type_() {
+                ChatType::Private(data) => {
+                    let blocked_expression = gtk::PropertyExpression::new(
+                        Chat::static_type(),
+                        Some(gtk::ConstantExpression::new(chat)),
+                        "is-blocked",
+                    )
+                    .upcast();
+                    let user_expression = gtk::ConstantExpression::new(data).upcast();
+                    let visible_expression = gtk::ClosureExpression::with_callback(
```

Comment:
Instead of using the "callback" closure expression I suggest using the regular ones (`gtk::ClosureExpression::new`). You can take a look on how it works [here](https://github.com/melix99/telegrand/blob/e64043f6b597d11c4f4078759ac359f14730b27a/src/session/sidebar/row.rs#L213).